test_framework/messages.py

```diff
@@ -381,6 +381,9 @@ def __repr__(self):
     def is_null(self):
         return self.scriptWitness.is_null()
 
+    def from_stack(self, witness_stack):
```

Comment:
Alternatively, you could just update `__init__()` to optionally take a stack.

Reply:
Agree, that's better. Updated.